tests/rptest/tests/leaders_info_api_test.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.rpk_producer import RpkProducer
+from rptest.tests.end_to_end import EndToEndTest
+
+from ducktape.utils.util import wait_until
+
+
+class LeadersInfoApiTest(EndToEndTest):
```

Comment:
Does this need to be a full `EndToEnd` test? I don't think so unless there is a specific reason

Reply:
I wanna generate traffic and during it reset leaders table and check that producer consumer make progress.
Looks like EndToEnd test is useful in this case.

Reply:
> I wanna generate traffic and during it reset leaders table and check that producer consumer make progress

This is great motivation for the test you wrote with EndToEnd test and perhaps you should keep the test.

But this particular feature is primarily geared towards scenarios where the system is in a bad state and we want to reset the cache to try and fix the bad state. So what is missing IMO is a test that tests precisely the cleaning of the metadata cache. That is independent of the behavior of clients / overall systems in such circumstances.

Reply:
I think this test check cleaning metadata. 
```
      def check_reset_leaders():
            for node in self.redpanda.nodes:
                self.admin.reset_leaders_info(node)

            partition_without_leader = 0
            for partition in range(self.spec.partition_count):
                leader = self.admin.get_partition_leader(namespace="kafka",
                                                         topic=self.topic,
                                                         partition=partition)
                if leader == -1:
                    partition_without_leader += 1

            return partition_without_leader >= 2

        wait_until(check_reset_leaders,
                   timeout_sec=180,
                   backoff_sec=1,
                   err_msg="Can not reset leaders_table for nodes")


```

Sorry( I think I do not understand